### PR TITLE
switch base image to python:alpine so latest version of ansible can be installed

### PR DESCRIPTION
**Pull request by eljohnson92**
Signed-off-by: Evan Johnson <eljohn1014@gmail.com>

### Description of your changes
The current base image used for this provider looks to be unmaintained for the past 3 years, this PR switches to using the official python:alpine base image and now installs both ansible and ansible-runner via pip install

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
I have test image is building successfully and both ansible and ansible-playbook are installed in the default PATH and on the latest version

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by fahedouch — CHANGES_REQUESTED** [review]

File: cluster/images/provider-ansible/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-FROM alpinelinux/ansible
-RUN apk --no-cache add ca-certificates bash
-RUN pip3 install ansible-runner
+FROM python:alpine
+RUN apk --no-cache add ca-certificates bash ansible-lint  openssh-client
```
[the remark this answers is not in the document]
Comment (eljohnson92) [reply cluster/images/provider-ansible/Dockerfile:2]: I was just trying to keep the image as similar to the alpinelinux/ansible image as possible, including the packages that it included. let me try removing some of those extra packages and see if things are still working

File: cluster/images/provider-ansible/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
-FROM alpinelinux/ansible
-RUN apk --no-cache add ca-certificates bash
-RUN pip3 install ansible-runner
+FROM python:alpine
```
[the remark this answers is not in the document]
Comment (eljohnson92) [reply cluster/images/provider-ansible/Dockerfile]: I've updated this and confirmed the image is working correctly

File: cluster/images/provider-ansible/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (eljohnson92) [reply cluster/images/provider-ansible/Dockerfile]: I have removed ansible-lint from the install here and confirmed the rest of the packages are necessary

File: cluster/images/provider-ansible/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (fahedouch) [reply cluster/images/provider-ansible/Dockerfile]: Thanks, we may use `ARG` (`build-arg`) in the future, but this can be done in another PR

**Comment by fahedouch** [comment]
Thanks @eljohnson92 .  I think we should add extra integration test on the GHA CI to test ansible provider setup within this docker images, [sthg like this](https://github.com/crossplane-contrib/provider-gcp/blob/master/cluster/local/integration_tests.sh). This can be done in an other PR too

**Pull request merged**